app/controllers/users_controller.rb

```diff
@@ -46,6 +46,35 @@ def check_account_creation_status
     end
   end
 
+  # Run a set of corrections on a user
```

Comment:
I think this needs moar comments -- up here it should at least give a vague overview of the problems we're fixing (like you did in the pull request description), and it wouldn't hurt to basically break the code up into paragraphs, with a comment before each one explaining, e.g. (and my apologies for including the +; I'm lazy)

```
 +    # Delete any subscriptions the user has to deleted items because this causes
 +    # the user's subscription page to error
 +    @subscriptions = @user.subscriptions.includes(:subscribable)
 +    @subscriptions.to_a.each do |sub| 
 +      if sub.name.nil? 
 +        sub.destroy
 +      end
 +    end 
 +
 +    # Do something with the revised at date for the user's works to fix something and then
 +    # reindex the user's works to make sure they show up on the user's works page
 +    @user.works.each do |work| 
 +      if work.revised_at.nil?
 +        work.save
 +      end
 +      IndexQueue.enqueue(w, :main)
 +    end
 +
 +    # comment
 +    @user.bookmarks.each do |bookmark|
 +      bookmark.update_index
 +    end
 +
 +    # comment
 +    @user.update_works_index_timestamp!
```